x-pack/plugins/rule_registry/server/routes/get_alerts_group_aggregations.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '@kbn/core/server';
+import * as t from 'io-ts';
+import { transformError } from '@kbn/securitysolution-es-utils';
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import Boom from '@hapi/boom';
+import { RacRequestHandlerContext } from '../types';
+import { BASE_RAC_ALERTS_API_PATH } from '../../common/constants';
+import { buildRouteValidation } from './utils/route_validation';
+import { alertsAggregationsSchema, alertsGroupFilterSchema } from '../../common/types';
+import { DEFAULT_ALERTS_GROUP_BY_FIELD_SIZE } from '../alert_data_client/constants';
+
+export const getAlertsGroupAggregations = (router: IRouter<RacRequestHandlerContext>) => {
+  router.post(
+    {
+      path: `${BASE_RAC_ALERTS_API_PATH}/_group_aggregations`,
+      validate: {
+        body: buildRouteValidation(
+          t.exact(
+            t.type({
+              featureIds: t.array(t.string),
+              groupByField: t.string,
+              aggregations: t.union([alertsAggregationsSchema, t.undefined]),
+              filters: t.union([t.array(alertsGroupFilterSchema), t.undefined]),
+              sort: t.union([t.array(t.object), t.undefined]),
+              pageIndex: t.union([t.number, t.undefined]),
+              pageSize: t.union([t.number, t.undefined]),
+            })
+          )
+        ),
+      },
+      options: {
+        tags: ['access:rac'],
+      },
+    },
+    async (context, request, response) => {
+      const {
+        featureIds,
+        groupByField,
+        aggregations,
+        filters,
+        sort,
+        pageIndex = 0,
+        pageSize = DEFAULT_ALERTS_GROUP_BY_FIELD_SIZE,
+      } = request.body;
+      if (pageIndex < 0 || pageIndex > 100) {
```

Comment:
With this validation, it seems that we cap users to not be able to go over the 100th page if the `pageSize` is `1`. What about the following?

```
if (pageIndex > maxPerPage) { <--- 100 items per page
  throw Boom.badRequest('The provided pageIndex value is too high. The maximum allowed pageIndex value is ${maxPerPage}.');
}

if (Math.max(pageIndex, pageIndex * pageSize) > MAX_DOCS_PER_PAGE) { <-- 10K max documents
   throw Boom.badRequest('The number of documents is too high. Paginating through more than ${MAX_DOCS_PER_PAGE} documents is not possible.');
}
```

Also, I am wondering if we should move the validation inside the `alertsClient.getGroupAggregations`. This way consumers of the alert client will have the same validation. Wdyt?